tests/test_scheduler.py

```diff
@@ -464,3 +472,80 @@ def test_lift_fields():
 
     result = lift_fields(plan)
     assert result == expected
+
+
+def test_normalize_names():
+    plan = Plan(
+        (
+            Field("##foo#8"),
+            Field("##foo#1"),
+            Field("#2#foo"),
+            Alias("##foo#9"),
+            Field("#10#A"),
+            Alias("bar"),
+            Field("j"),
+            Alias("##test#0"),
+        )
+    )
+
+    expected = Plan(
+        (
+            Field("foo_2"),
+            Field("foo_3"),
+            Field("foo_4"),
+            Alias("foo"),
+            Field("A"),
+            Alias("bar"),
+            Field("j"),
+            Alias("test"),
+        )
+    )
+
+    result = normalize_names(plan)
+    assert result == expected
+
+
+class HashableEagerTensor(TestEagerTensor):
+    # TODO: @mtsokol This is dreadful but it's only for testing
+    #       purposes. The plan that we pass to scheduler needs
+    #       be hashable, as well arrays that are within.
```

Comment:
Do we think that we should set the hash for Immediate to use the id of the object?

Reply:
I think that makes sense.